### DISCUSSION TITLE
`null` json body is considered bad request by ServerRequestInterface?

### DISCUSSION THREAD

**Discussion by nanaya** [laravel/framework#53511] [Q&A]
1. clone [null-json-body](<https://github.com/nanaya/laravel/tree/null-json-body>) test branch
2. run `composer install`
3. init database with `php artisan migrate --force -n`
4. start server with `php artisan serve`
6. on another terminal, confirm the app runs correctly:
   ```
   $ curl -H 'Accept: application/json' -H 'Content-Type: application/json' --data '{"hello":"world"}' 127.0.0.1:8000; echo
   {"request":{"hello":"world"}}
   ```
7. run the same request but the body being null:
   ```
   $ curl -H 'Accept: application/json' -H 'Content-Type: application/json' --data 'null' 127.0.0.1:8000; echo
   {
       "message": "Bad request."
   }
   ```

Why is it returning error? `null` json body should be a valid request from what I can tell.

(I'm asking this because [sentry-laravel](https://github.com/getsentry/sentry-laravel/blob/master/src/Sentry/Laravel/Http/SetRequestMiddleware.php) is using this specific interface to generate report. it handles the error but it also means an exception is thrown every time such request is made)

**Comment by ismlunal** [comment]
❤

**Comment by ghost** [comment]
null is not a valid json.

Update: But according to this it is..
https://stackoverflow.com/questions/8526995/is-null-valid-json-4-bytes-nothing-else
